### PR TITLE
Add requirement Section in README and Fix Makefile

### PR DESCRIPTION
**Pull request by iAmG-r00t**
- Added a requirement section for a new user.
- The makefile generated contained a wrong command: `md5` instead of `md5sum`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,6 +12,18 @@ A Game Boy ROM disassembler.
 - Uses defines from hardware.inc v2.7 for hardware registers ([source](https://github.com/tobiasvl/hardware.inc))
 - Slow on large ROMs
 
+## Requirements
+
+To be able to use this tool you will require to install the [Rendex Game Boy Development System (RGBDS)](https://github.com/gbdev/rgbds) tool.
```
Comment (ISSOtm) [inline README.md:17]:
```suggestion
The assembly files generated by mgbdis are designed to be assembled with [RGBDS](https://rgbds.gbdev.io).
```

I think this phrasing is a little clearer. Additionally, it's better to link to the official website, as it contains per-OS install instructions—`make install` won't work on Windows out of the box, for example, and other OSes don't use `apt-get` and may require different packages.

File: mgbdis.py [hunk 2]
```diff
@@ -1170,7 +1170,7 @@ def write_makefile(self):
         else:
             f.write('\trgblink -n game.sym -m game.map -o $@ $<\n')
         f.write('\trgbfix -v -p 255 $@\n\n')
-        f.write('\tmd5 $@\n\n')
+        f.write('\tmd5sum $@\n\n')
```
Comment (ISSOtm) [inline mgbdis.py:1173]: This is correct on my Linux box, but @mattcurrie might be able to confirm whether that's the case on macOS?

**Review by ISSOtm — CHANGES_REQUESTED** [review]

File: mgbdis.py (hunk 2 above)
Comment (iAmG-r00t) [reply mgbdis.py:1173]: Yep on macOS to get md5 HASH one would use `md5`, a fix could be a function to check the OS if it's Linux use `md5sum` else if its macOS use `md5`

Comment (ISSOtm) [reply mgbdis.py:1173]: Alternatively, `if which md5sum &>/dev/null; then md5sum $@; else md5 $@; fi` to detect the presence of the tool itself regardless of OS.

Comment (mattcurrie) [reply mgbdis.py:1173]: Yeah, I prefer the detection approach as it makes the makefile portable

Comment (iAmG-r00t) [reply mgbdis.py:1173]: Okay found a solution, let me update then @mattcurrie you check if it works properly on macOS

Comment (iAmG-r00t) [reply mgbdis.py]: @ISSOtm kindly review and check if the changes are okay and @mattcurrie check if the command detection is portable.

Comment (mattcurrie) [reply mgbdis.py]: @iAmG-r00t I've just tested the detection command in the makefile and it works fine for me on macOS.

Comment (iAmG-r00t) [reply mgbdis.py]: Okay then you should proceed and push.

**Review by ISSOtm — APPROVED** [review]

File: mgbdis.py [hunk 3]
```diff
@@ -1170,7 +1171,10 @@ def write_makefile(self):
         else:
             f.write('\trgblink -n game.sym -m game.map -o $@ $<\n')
         f.write('\trgbfix -v -p 255 $@\n\n')
-        f.write('\tmd5 $@\n\n')
+        if which("md5") is not None:
+            f.write('\tmd5 $@\n\n')
+        else:
+            f.write('\tmd5sum $@\n\n')
```
Comment (mattcurrie) [inline mgbdis.py:1177]:
```suggestion
        f.write('\t@if which md5sum &>/dev/null; then md5sum $@; else md5 $@; fi\n\n')
```

File: mgbdis.py [hunk 4]
```diff
@@ -14,6 +14,7 @@
 import os
 import png
 from shutil import copyfile
+from shutil import which
```
Comment (mattcurrie) [inline mgbdis.py:17]:
```suggestion
```

**Comment by mattcurrie** [comment]
Thanks @iAmG-r00t and @ISSOtm

**Pull request merged**